service/object.go

```diff
@@ -1006,7 +1006,8 @@ type PutObjectInput struct {
 	// Specify the storage class for object
 	// XQSStorageClass's available values: STANDARD, STANDARD_IA
 	XQSStorageClass *string `json:"X-QS-Storage-Class,omitempty" name:"X-QS-Storage-Class" location:"headers"`
-
+	// User-defined metadata
+	XQSMetaData *map[string]string `json:"X-QS-MetaData,omitempty" name:"X-QS-MetaData" location:"headers"`
```

Comment:
We should not update here manually, is API Spec updated?

Reply:
I still don't think it's a good idea to add `x-qs-metadata` in specs directly, because we don't send them directly.

Maybe we can add following functions: 

`PutObjectInput.SetMetadata(key, value string)`
`PutObjectInput.GetMetadata(key string)`

If you think current implemention is good enough, I will ACK current PR and leave an issue for this.

Reply:
API Spec has updated, and the `PutObjectInput` is generated by snips .  I think your way
> Update template for sdk, add XQSMetaData as a map into PutObjectInput

is better for its consistency with other data in struct `PutObjectInput` than another way. So I think maybe it is OK.